### PR TITLE
jinja: reimplement auth template variables (bug 1908137)

### PR DESCRIPTION
**Pull request by zzzeid**
- reimplement user_has_phabricator_token
- reimplement user_is_authenticated (originally is_user_authenticated)

### CODE REVIEW

File: src/lando/ui/jinja2/partials/layout.html [hunk 1]
```diff
@@ -1,3 +1,11 @@
+{% if request %}
```
Comment (cgsheeh) [inline src/lando/ui/jinja2/partials/layout.html:1]: I'm very surprised there is no way to add our own custom variables/functions/etc to templates via Python.

**Review by cgsheeh — APPROVED** [review]

File: src/lando/ui/jinja2/partials/layout.html (hunk 1 above)
Comment (zzzeid) [reply src/lando/ui/jinja2/partials/layout.html:1]: > I'm very surprised there is no way to add our own custom variables/functions/etc to templates via Python.

There is, it's just that the request context is not readily available for those variables/functions. Technically you could implement a method that gets passed the request and call it here (e.g., `user_is_authenticated = check_if_user_is_authenticated(request)`).

In the case of `user_is_authenticated`, the variable is very simple and more so used like an alias, and doesn't really need a separate method.

Comment (zzzeid) [reply src/lando/ui/jinja2/partials/layout.html:1]: I guess implementing that method, even if it seems kind of redundant, will remove the need to include an if statement in the template. So upon rethinking this, I think it's OK to do that here.

Comment (zzzeid) [reply src/lando/ui/jinja2/partials/layout.html:1]: Filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1909734) to track the improvement to this.

**Pull request merged**